### ISSUE TITLE
passing embeddings to loss function

### ISSUE THREAD

**Issue by annongitter** [KevinMusgrave/pytorch-metric-learning#316]
According to the docs, the standard way to initialise ArcfaceLoss looks something like so:

```
loss_func = losses.ArcFaceLoss(...).to(torch.device('cuda'))
loss_optimizer = torch.optim.SGD(loss_func.parameters(), lr=0.01)
## get loss by passing embeddings to loss_func
# then during training:
loss_optimizer.step()
```

So, when the embeddings (say dim 512) are passed to the loss_func, do the embeds need to be normed? I see that the embeddings are normed here (but iam not entirely sure this is done in arcfaceloss?):

`https://github.com/KevinMusgrave/pytorch-metric-learning/blob/dff4ae570db89dcb59a102f13f665502f9c1c7c6/src/pytorch_metric_learning/losses/large_margin_softmax_loss.py#L87`

So, we just send the raw embeds to the loss function?
Thank you!

**Comment by KevinMusgrave** [comment]
The embeddings do not need to be normalized. They are getting normalized here:

https://github.com/KevinMusgrave/pytorch-metric-learning/blob/dff4ae570db89dcb59a102f13f665502f9c1c7c6/src/pytorch_metric_learning/losses/large_margin_softmax_loss.py#L56-L57

because the default distance type is CosineSimilarity, which normalizes embeddings.
